content/en/docs/reference/networking/virtual-ips.md

```diff
@@ -0,0 +1,301 @@
+---
+title: Virtual IPs and Service Proxies
+content_type: reference
+---
+
+<!-- overview -->
+Every {{< glossary_tooltip term_id="node" text="node" >}} in a Kubernetes
+cluster runs a [kube-proxy](/docs/reference/command-line-tools-reference/kube-proxy/)
+(unless you have deployed your own alternative component in place of `kube-proxy`).
+
+The `kube-proxy` component is responsible for implementing a _virtual IP_
+mechanism for {{< glossary_tooltip term_id="service" text="Services">}}
+of `type` other than
+[`ExternalName`](/docs/concepts/services-networking/service/#externalname).
+
+<!-- body -->
+
+## Motivation
+
+A question that pops up every now and then is why Kubernetes relies on
+proxying to forward inbound traffic to backends. What about other
+approaches? For example, would it be possible to configure DNS records that
+have multiple A values (or AAAA for IPv6), and rely on round-robin name
+resolution?
+
+There are a few reasons for using proxying for Services:
+
+* There is a long history of DNS implementations not respecting record TTLs,
+  and caching the results of name lookups after they should have expired.
+* Some apps do DNS lookups only once and cache the results indefinitely.
+* Even if apps and libraries did proper re-resolution, the low or zero TTLs
+  on the DNS records could impose a high load on DNS that then becomes
+  difficult to manage.
+
+Later in this page you can read about various kube-proxy implementations work. Overall,
+you should note that, when running `kube-proxy`, kernel level rules may be modified
+(for example, iptables rules might get created), which won't get cleaned up, in some
+cases until you reboot.  Thus, running kube-proxy is something that should only be done
+by an administrator which understands the consequences of having a low level, privileged
+network proxying service on a computer.  Although the `kube-proxy` executable supports a
+`cleanup` function, this function is not an official feature and thus is only available
+to use as-is.
+
+### Configuration
+
+Note that the kube-proxy starts up in different modes, which are determined by its configuration.
+
+- The kube-proxy's configuration is done via a ConfigMap, and the ConfigMap for
+  kube-proxy effectively deprecates the behaviour for almost all of the flags for
+  the kube-proxy.
+- The ConfigMap for the kube-proxy does not support live reloading of configuration.
+- The ConfigMap parameters for the kube-proxy cannot all be validated and verified
+  on startup.  For example, if your operating system doesn't allow you to run iptables
+  commands, the standard kernel kube-proxy implementation will not work.  Likewise,
+  if you have an operating system which doesn't support `netsh`, it will not run in
+  Windows userspace mode.
+
+
+### User space proxy mode {#proxy-mode-userspace}
```

Comment:
This has been deprecated and temptatively  removed in 1.25
https://github.com/kubernetes/kubernetes/pull/104631

Reply:
This is a refactor PR; we should do the docs update for the v1.23 release separately. I'd hold to that even though the docs for the proxy mode are late. One reason is that refactor PRs of this size can take between one and nine months to merge, which is a long time to wait for a docs bugfix.

I've filed https://github.com/kubernetes/website/issues/32421 to track that bug.

Either this PR will merge first, and that work will still need doing, or that PR will merge first and I can rebase this on top of the updated _main_ branch.

Reply:
I think this is resolved.

BTW: am I right that the user-space proxy support is till there and overdue for removal?